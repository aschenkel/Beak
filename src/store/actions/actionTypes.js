export const SET_USER = 'SET_USER';
export const SET_PROFILE = 'SET_PROFILE';
export const SET_TWEETS = 'SET_TWEETS';
export const ERROR_USER_LOGIN = 'ERROR_USER_LOGIN';
export const LOG_OUT = 'LOG_OUT';
export const INIT_COMPLETED = 'INIT_COMPLETED';