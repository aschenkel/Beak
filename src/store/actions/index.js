import * as actionTypes from './actionTypes'
import FabricTwitterKit from 'react-native-fabric-twitterkit'
import { AsyncStorage } from 'react-native';
var Twitter = require('twitter-node-client').Twitter;
var config = require('../../../twitter_config.json');

export const setUser = (session) => ({
    type: actionTypes.SET_USER,
    userName: session.userName,
    userID: session.userID,
    authorized:true,
    authToken: session.authToken,
    authTokenSecret: session.authTokenSecret,
    error:false
})

export const signIn = () => {
    return function (dispatch) {
        return FabricTwitterKit.login((error, res) => 
        {
            if(res !== undefined){
                if(res.userName){
                    dispatch(setUser(res))
                    AsyncStorage.setItem('session', JSON.stringify(res)) 
                }else{
                    dispatch(errorLoadingUser()); 
                }
            }
        });
    };
}

export const loadUser = () => {
    return function (dispatch) {
        AsyncStorage.getItem('session').then((session) => {
            if (session) {
                dispatch(setUser(JSON.parse(session)))
            }
        })
        dispatch(initCompleted())
    };
}

export const fetchTweets = () => {
    return function (dispatch,getState) {
        var twitter = new Twitter(config);
        var error = (err, response, body) =>{
            //dispatch error
        };
        var success =  response => {
            var data = JSON.parse(response)
            dispatch(setTweets(data))
        };    
        twitter.getUserTimeline({ user_id: ""+ getState().user.userID, count: '30',include_rts:'false'}, error, success)
    };
}

export const logOut = () => {
    return function (dispatch,getState) {
        FabricTwitterKit.logOut() //Doesnt work. Issue: https://github.com/tkporter/react-native-fabric-twitterkit/issues/33
        AsyncStorage.removeItem('session')
        dispatch(logOutAction())
    };
}

export const fetchProfile= () => {
    return function (dispatch) {
        FabricTwitterKit.fetchProfile((error, profile) => 
            {
                if(profile !== undefined){
                    dispatch(setProfile(profile))
                }
            })
    };
}

export const setProfile = (profile) => ({
    type: actionTypes.SET_PROFILE,
    name: profile.name,
    imageURL: profile.profile_image_url_https
})

export const logOutAction = () => ({
    type: actionTypes.LOG_OUT,
})

export const setTweets = (tweets) => ({
    type: actionTypes.SET_TWEETS,
    tweets: tweets,
})

export const errorLoadingUser = () => ({
  type: actionTypes.ERROR_USER_LOGIN,
  error:true
})

export const initCompleted = () => ({
  type: actionTypes.INIT_COMPLETED,
  init:true
})

