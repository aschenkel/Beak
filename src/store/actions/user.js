import * as actionTypes from './actionTypes'
import FabricTwitterKit from 'react-native-fabric-twitterkit'
import { AsyncStorage } from 'react-native';

export const setUser = (session) => ({
    type: actionTypes.SET_USER,
    userName: session.userName,
    userID: session.userID,
    authorized:true,
    authToken: session.authToken,
    authTokenSecret: session.authTokenSecret,
    error:false
})

export const signIn = (api = FabricTwitterKit) => {
    return function (dispatch) {
        return api.login((error, res) => 
        {
            if(res !== undefined){
                if(res.userName){
                    dispatch(setUser(res))
                    AsyncStorage.setItem('session', JSON.stringify(res)) 
                }else{
                    dispatch(errorLoadingUser()); 
                }
            }
        });
    };
}

export const loadUser = () => {
    return function (dispatch) {
        AsyncStorage.getItem('session').then((session) => {
            if (session) {
                dispatch(setUser(JSON.parse(session)))
            }
        })
        dispatch(initCompleted())
    };
}

export const logOut = () => {
    return function (dispatch,getState) {
        FabricTwitterKit.logOut() //Doesnt work. Issue: https://github.com/tkporter/react-native-fabric-twitterkit/issues/33
        AsyncStorage.removeItem('session')
        dispatch(logOutAction())
    };
}

export const fetchProfile= (api = FabricTwitterKit) => {
    return function (dispatch) {
        api.fetchProfile((error, profile) => 
            {
                if(profile !== undefined){
                    dispatch(setProfile(profile))
                }
            })
    };
}

export const setProfile = (profile) => ({
    type: actionTypes.SET_PROFILE,
    name: profile.name,
    imageURL: profile.profile_image_url_https
})

export const logOutAction = () => ({
    type: actionTypes.LOG_OUT,
})

export const errorLoadingUser = () => ({
  type: actionTypes.ERROR_USER_LOGIN,
  error:true
})

export const initCompleted = () => ({
  type: actionTypes.INIT_COMPLETED,
  init:true
})

