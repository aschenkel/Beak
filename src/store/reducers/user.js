import * as actionTypes from '../actions/actionTypes'

var initialState = {
    init:false,
    authorized: false,
    userName: '',
    userID: '',
    authToken: '',
    authTokenSecret: '',
    error: false
}

const user = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.INIT_COMPLETED:
      return {
        ...state,
        init:true
      }
    case actionTypes.ERROR_USER_LOGIN:
      return {
        ...state,
        error:true
      }
    case actionTypes.LOAD_USER:
      return {
        ...state,
        userName: action.userName,
        userID: action.userID,
        authorized:true,
        authToken: action.authToken,
        authTokenSecret: action.authTokenSecret,
        error:false
      }
    default:
      return state
  }
}

export default user