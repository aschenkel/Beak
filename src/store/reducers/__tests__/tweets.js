import { tweets } from '../tweets';
import { initialState } from '../user';
import { setTweets } from '../../actions/tweets';

it('SetTweets: should update state setting users tweets', () => {
  var tweetsData = [{text:"this is a tweet"},{text:"just another tweet"}, {text:"a third one, yep"}]
  const newState = tweets(initialState, setTweets(tweetsData));
  expect(newState).toMatchSnapshot();
});