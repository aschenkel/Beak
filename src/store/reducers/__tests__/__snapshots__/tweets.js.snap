// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetTweets: should update state setting users tweets 1`] = `
Object {
  "authToken": "",
  "authTokenSecret": "",
  "authorized": false,
  "error": false,
  "grades": Array [
    2,
    4,
    2,
  ],
  "imageURL": null,
  "init": false,
  "name": "",
  "tweets": Array [
    Object {
      "text": "this is a tweet",
    },
    Object {
      "text": "just another tweet",
    },
    Object {
      "text": "a third one, yep",
    },
  ],
  "tweetsReady": true,
  "userID": "",
  "userName": "",
}
`;
